src/picongpu/include/plugins/adios/ADIOSWriter.hpp

```diff
@@ -494,29 +625,100 @@ class ADIOSWriter : public ILightweightPlugin
         ADIOS_CMD(adios_init_noxml(mThreadParams.adiosComm));
     }
 
+    /**
+     * Notification for dump or checkpoint received
+     *
+     * @param currentStep current simulation step
+     * @param isCheckpoint checkpoint notification
+     */
+    void notificationReceived(uint32_t currentStep, bool isCheckpoint)
+    {
+        const PMacc::Selection<simDim>& localDomain = Environment<simDim>::get().SubGrid().getLocalDomain();
+        mThreadParams.isCheckpoint = isCheckpoint;
+        mThreadParams.currentStep = currentStep;
+        mThreadParams.cellDescription = this->cellDescription;
+        this->filter.setStatus(false);
+
+        __getTransactionEvent().waitForFinished();
+
+        std::string fname = filename;
+        if (isCheckpoint)
+        {
+            /* if checkpointFilename is relative, prepend with checkpointDirectory */
+            if (!boost::filesystem::path(checkpointFilename).has_root_path())
+                fname = checkpointDirectory + std::string("/") + checkpointFilename;
+            else
+                fname = checkpointFilename;
+
+            mThreadParams.window = MovingWindow::getInstance().getDomainAsWindow(currentStep);
+        }
+        else
+        {
+            mThreadParams.window = MovingWindow::getInstance().getWindow(currentStep);
+        }
+
+        for (uint32_t i = 0; i < simDim; ++i)
+        {
+            mThreadParams.localWindowToDomainOffset[i] = 0;
+            if (mThreadParams.window.globalDimensions.offset[i] > localDomain.offset[i])
+            {
+                mThreadParams.localWindowToDomainOffset[i] =
+                    mThreadParams.window.globalDimensions.offset[i] -
+                    localDomain.offset[i];
+            }
+        }
+
+        if (MovingWindow::getInstance().isSlidingWindowActive())
```

Comment:
This if should only set the filter if `isCheckpoint==false`. IMO the full if statement must moved to the else path in line `657`